db/bbolt.go

```diff
@@ -64,80 +64,68 @@ func (bb *BBoltHandler) CreateInitialSchema() error {
 	})
 }
 
-func (bb *BBoltHandler) InsertBlock(block *types.Block) error {
-	bb.logger.Info("Inserting block to DB", zap.Uint64("block_height", block.BlockHeight))
-
-	// Store mapping number -> block
-	err := bb.db.Update(func(tx *bolt.Tx) error {
-		b := tx.Bucket([]byte(blocksBucket))
-		key := bb.itob(block.BlockHeight)
-		blockBytes, err := json.Marshal(block)
-		if err != nil {
-			return err
-		}
-		return b.Put(key, blockBytes)
-	})
-	if err != nil {
-		bb.logger.Error("Error inserting block", zap.Error(err))
-		return err
+func (bb *BBoltHandler) InsertBlocks(blocks []*types.Block) error {
+	if len(blocks) == 0 {
+		return nil
 	}
 
-	// Store mapping hash -> number
-	err = bb.db.Update(func(tx *bolt.Tx) error {
-		b := tx.Bucket([]byte(blockHeightsBucket))
-		return b.Put([]byte(block.BlockHash), bb.itob(block.BlockHeight))
-	})
-	if err != nil {
-		bb.logger.Error("Error inserting block", zap.Error(err))
-		return err
-	}
+	bb.logger.Info("Batch inserting blocks to DB", zap.Int("count", len(blocks)))
+
+	// Single transaction for all operations
+	return bb.db.Update(func(tx *bolt.Tx) error {
+		blocksBucket := tx.Bucket([]byte(blocksBucket))
+		heightsBucket := tx.Bucket([]byte(blockHeightsBucket))
+		indexBucket := tx.Bucket([]byte(indexerBucket))
+
+		var minHeight, maxHeight uint64 = math.MaxUint64, 0
+
+		// Insert all blocks
+		for _, block := range blocks {
+			// Update min/max heights
+			if block.BlockHeight < minHeight {
+				minHeight = block.BlockHeight
+			}
+			if block.BlockHeight > maxHeight {
+				maxHeight = block.BlockHeight
+			}
+
+			// Store block data
+			blockBytes, err := json.Marshal(block)
+			if err != nil {
+				return err
```

Comment:
we used to have error logs such as https://github.com/babylonlabs-io/finality-gadget/blob/37a8b1c9b2698c9967bbc060583668d6c7a1e6f9/db/bbolt.go#L81

can we add them back?